src/pygambit/game.pxi

```diff
@@ -768,10 +773,50 @@ class Game:
             if not rational:
```

Comment:
In this case, now that so much code has been added to this branch of the `if self.is_tree`, it would be much better to reverse the conditional.  That is, have a conditional that checks `if not self.is_tree` first.  Then, you do not need an `else` after that.  This can be seen by imitating the implementation of `mixed_strategy_profile` which is immediately adjacent in the source file.